lammps-tutorials.tex

```diff
@@ -181,257 +306,573 @@ \section{Prerequisites}
-input script. For clarity, the input scripts for this tutorial will be divided
-into five categories, which we will fill out one by one. First, create a folder
-named \textit{my-first-input/}, and then create a blank text file in it called
-\textit{input.lmp}. Open \textit{input.lmp} using the LAMMPS-GUI. An empty
-buffer should open, into which you can copy-paste the following lines:
-{\normalsize \begin{verbatim}
-# PART A - ENERGY MINIMIZATION
-# 1) Initialization
-# 2) System definition
-# 3) Settings
-# 4) Visualization
-# 5) Run
-\end{verbatim}}
-A line starting with a hash symbol ($\#$) is a comment ignored by LAMMPS. These
-five categories are not required in every input script and do not necessarily
-need to be in that exact order. For instance, parts 3 and 4 could be inverted,
-or part 4 could be omitted. Note, however, that LAMMPS reads input files from
-top to bottom. Therefore, the \textit{Initialization} and \textit{System
-definition} categories must appear at the top of the input, and the \textit{Run}
-category at the bottom.
+To run a simulation using LAMMPS, you need to write an input script with
+series of commands of commands that LAMMPS will then execute.  For
+clarity, the input scripts for this tutorial will be divided into five
+categories, which we will fill out step by step.  First, create a folder
+named \textit{my-first-input/} for all files of this tutorial, then
+launch LAMMPS--GUI, copy-n-paste the text below into the \textit{Editor}
+window, and save it to the folder as a file called \textit{input.lmp}.
+{\normalsize
+\begin{verbatim}
+ # PART A - ENERGY MINIMIZATION
+ # 1) Initialization
+ # 2) System definition
+ # 3) Settings
+ # 4) Visualization
+ # 5) Run
+\end{verbatim}
+}
+
+A line or part of a line starting with a hash symbol ($\#$) is a
+comment ignored by LAMMPS.  LAMMPS--GUI will color such comments in red.
+These five categories are not required in every input script and do not
+necessarily need to be in that exact order.  For instance, parts 3 and 4
+could be inverted, or part 4 could be omitted.  Note, however, that
+LAMMPS reads input files from top to bottom and processes each command
+in it \emph{immediately}.  Therefore, the \textit{Initialization} and
+\textit{System definition} categories must appear at the top of the
+input, and the \textit{Run} category at the bottom.  Also, the specifics
+of some commands can change after global settings are changed, so the
+order of commands in the input script matters for such cases.
+
+The contents of the \textit{Editor} window can be saved by either using
+the ``Save'' or ``Save As'' entry in the ``File'' menu, using the
+\texttt{Ctrl--S} keyboard shortcut or by clicking on the ``Save'' icon
+at the bottom left of the \textit{Editor} window status bar.  For
+running LAMMPS in LAMMPS--GUI it is not required to save the buffer.
+The current contents of the buffer will be passed on to LAMMPS.
```

Comment:
One of the earliest design choices for LAMMPS-GUI was to run directly from the content of the edit buffer, i.e. to get away from the "save then run" setup required with other editors. The rationale being that LAMMPS-GUI would be used for quick experimenting. Since almost everything can be done with keyboard shortcuts, saving the content of the buffer can be quick (Ctrl-S and Ctrl-Enter) will do what you expect.

But I can see how this would be unexpected for others. I don't want to change the default behavior, but I think an "auto-save" could be added as an optional feature. One could provide an "idle interval" or save on "run" or "quit". I'll add this to my TODO list.

Reply:
FYI, an auto-save or "run" and "quit" has been implemented in commit https://github.com/lammps/lammps/pull/4255/commits/d470919a4424e4cb618b4df9560a442ae6f909d3
I need to update the docs and then will push new binaries to the testing download server tonight.